scraping/scraping/start_scripts/amazon_eu.py

```diff
@@ -0,0 +1,272 @@
+
+# unfortunately the json file only contains leaf nodes
+# so we have to do some extra work to map a browse node path to its leaf nodes.
+# we wouldnt need to do that if we had a list of internal nodes aswell.
+
+
+def find_leaf_nodes_by_path(path: str) -> Tuple[int, int]:
+    """
+    find all leaf nodes matching a given path
+
+    Args:
+        path (str): the name of some amazon browse node
+
+    Returns:
+        Tuple[int, int]: the range of leaf nodes given as (length, offset)
+    """
+    seperator = "/"
+    assert not path.startswith(seperator) and not path.endswith(seperator)
+    one_after_seperator = chr(ord(seperator) + 1)
+    start = bisect_left(browse_tree_leaves, path + seperator, key=sort_key)
+    end = bisect_left(browse_tree_leaves, path + one_after_seperator, key=sort_key)
+    return end - start, start
+
+
+def translate_paths_in_category_map(path_2_category: dict) -> dict:
+    """
+    replaces each path in a category mapping with the corresponding leaf nodes.
+    here its more convenient to refer to a leaf node by its index.
+
+    Args:
+        path_2_category (dict): a map from paths to categories
+
+    Returns:
+        dict: a map from leaf node indices to categories
+    """
+
+    ranges = sorted(
+        (*find_leaf_nodes_by_path(browse_node), category)
+        for browse_node, category in path_2_category.items()
+    )
+
+    index_2_category = {}
+
+    for length, offset, category in reversed(ranges):
+        for i in range(offset, offset + length):
+            index_2_category[i] = category
+
+    return index_2_category
+
+
+def combine_results(
+    country_code: str,
+    path_2_category: dict,
+    metadata: dict,
+    gender: Optional[str] = None,
+    consumer_lifestage: Optional[str] = None,
+) -> List[dict]:
+
+    country_code_to_url = {
+        "fr": "https://www.amazon.fr",
+        "de": "https://www.amazon.de",
+        "uk": "https://www.amazon.co.uk",
+        "it": "https://www.amazon.it",
+        "es": "https://www.amazon.es",
+    }
+
+    base_url = country_code_to_url[country_code]
+
+    index_2_category = translate_paths_in_category_map(path_2_category)
+
+    results = []
+    for i, category in index_2_category.items():
+        if category:
+            if node := browse_tree_leaves[i]["id"][country_code]:
+                results.append(
+                    {
+                        "start_urls": f"{base_url}/s?bbn={node}&rh=n%3A{node}%2Cp_n_cpf_eligible%3A22579881031",  # noqa
+                        "category": category,
+                        "gender": gender,
+                        "consumer_lifestage": consumer_lifestage,
+                        "meta_data": json.dumps(metadata),
+                    }
+                )
+    return results
+
+
+def female(country_code: str) -> List[dict]:
+    path_2_category = {
+        "uk-apparel/Categories/Women/Blouses & Shirts": "BLOUSE",
```

Comment:
I just tried a few links and amazon says no results (at least climate pledge-friendly results): 

```
"Showing results from All Departments
No results for Men's Boxer Shorts in Clothing"
```

These are the links I tried:

[{'start_urls': 'https://www.amazon.de/s?bbn=1981401031&rh=n%3A1981401031%2Cp_n_cpf_eligible%3A22579881031',
 {'start_urls': 'https://www.amazon.de/s?bbn=1981402031&rh=n%3A1981402031%2Cp_n_cpf_eligible%3A22579881031',
 {'start_urls': 'https://www.amazon.de/s?bbn=2153286031&rh=n%3A2153286031%2Cp_n_cpf_eligible%3A22579881031',

I also checked uk and the same problem occurred. Can you check it, please?